examples/kafka_spark_demo/feature_repo/streaming_features_tutorial.ipynb

```diff
@@ -0,0 +1,877 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Building streaming features with Feast, Spark, and Kafka"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Overview\n",
+    "\n",
+    "This notebook explores how data scientists and engineers can build streaming features in Feast. It builds off of the first [module](https://github.com/feast-dev/feast-workshop/tree/main/module_1) from the Feast workshop. The workshop module is not a prerequisite, but is recommended if you are not already familiar with the Push API in Feast.\n",
+    "\n",
+    "A typical pattern that we see is that data scientists use Feast to define, use, and share features, which greatly improves their productivity, while ML engineers use Feast to ensure that all the features are available for both training and serving. This notebook will first show how a data scientist can define and use stream features in Feast, and then how an ML engineer can use the Push API to ensure that those streaming features are available in production.\n",
+    "\n",
+    "All the necessary resources to run this notebook are in `kafka_spark_demo`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Setup the feature store"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Apply feature repository\n",
+    "We first run `feast apply` to register the data sources and features."
```

Comment:
IMO it is a bit strange that we start this demo jumping right into `feast apply`. I'd have a little bit more context about the challenges and opportunities with streaming in the introduction section.

Reply:
added some exposition